src/sync/installation.ts

```diff
@@ -406,22 +417,36 @@ export const processInstallation =
 				}
 
 				// Continue sync when a 404/NOT_FOUND is returned
-				if (handleNotFoundErrors(err, job, nextTask)) {
+				if (isNotFoundError(err, job, nextTask)) {
 					const edgesLeft = []; // No edges left to process since the repository doesn't exist
 					await updateJobStatus(queues, job, edgesLeft, task, repositoryId);
 					return;
 				}
 
-				await subscription.update({ syncStatus: "FAILED" });
+				if (await booleanFlag(BooleanFlags.CONTINUE_SYNC_ON_ERROR, false, jiraHost)) {
+
+					logger.warn({ job, task: nextTask, err }, "Task failed, continuing with next task");
 
-				const host = subscription.jiraHost || "none";
-				logger.warn({ job, task: nextTask, err, jiraHost: host }, "Sync failed");
+					// marking the current task as failed
+					await subscription.updateRepoSyncStateItem(nextTask.repositoryId, getStatusKey(nextTask.task as TaskType), "failed");
 
-				job.sentry.setExtra("Installation FAILED", JSON.stringify(err, null, 2));
-				job.sentry.captureException(err);
+					statsd.increment(metricTaskStatus.failed, [`type: ${nextTask.task}`]);
 
-				statsd.increment(metricSyncStatus.failed);
+					// queueing the job again to pick up the next task
+					queues.installation.add(job.data);
+				} else {
```

Comment:
This is the main change. In case of an unhandled error, we now update the status of the task as "failed" and schedule another message on the queue to continue with the next task.